File: src/lib/pjecalc/core/comum/validators/greater-or-equal-than-validator.ts
/**
 * PJe-Calc v2.15.1 — GreaterOrEqualThanValidator
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.GreaterOrEqualThanValidator
 */
import { HelperDate } from '../../base/comum/helper-date';
import { CustomValidator } from './custom-validator';
import type { GreaterOrEqualThan_ } from './greater-or-equal-than';
import type { ValidatorContext } from './validator-context';

export class GreaterOrEqualThanValidator extends CustomValidator<GreaterOrEqualThan_> {
  private static readonly COMPARED_KEY = '1';

  getMessage(): string { return this.spec.message; }
  getGroups(): number[] { return this.spec.groups; }
  protected getCondition(): string { return this.spec.condition; }

  getParameter(key: string): unknown {
    if (key === GreaterOrEqualThanValidator.COMPARED_KEY) return this.spec.value;
    return null;
  }

  isValid(context: ValidatorContext): boolean {
    const value1 = context.getValue();
    const value2 = context.getMemberValue(this.spec.value);
    if (value1 == null || value2 == null) return true;
    if (value1 instanceof Date && value2 instanceof Date) {
      return HelperDate.dateBeforeOrEquals(value2, value1);
    }
    return true;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/GreaterOrEqualThanValidator.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.base.comum.HelperDate;
import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.base.comum.validadores.CustomValidator;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.GreaterOrEqualThan;
import java.util.Date;

public class GreaterOrEqualThanValidator
extends CustomValidator<GreaterOrEqualThan> {
    private static final String COMPARED_KEY = "1";

    @Override
    public Object getParameter(String key) {
        if (key.equals(COMPARED_KEY)) {
            return ((GreaterOrEqualThan)this.getAnnotation()).value();
        }
        return null;
    }

    @Override
    public String getMessage() {
        return ((GreaterOrEqualThan)this.getAnnotation()).message();
    }

    @Override
    public byte[] getGroups() {
        return ((GreaterOrEqualThan)this.getAnnotation()).groups();
    }

    @Override
    protected String getCondition() {
        return ((GreaterOrEqualThan)this.getAnnotation()).condition();
    }

    @Override
    public boolean isValid(ValidatorContext context) {
        Object value1 = context.getValue();
        Object value2 = context.getMemberValue(((GreaterOrEqualThan)this.getAnnotation()).value());
        if (Utils.nulo(value1) || Utils.nulo(value2)) {
            return true;
        }
        return HelperDate.dateBeforeOrEquals((Date)value2, (Date)value1);
    }
}

